ui-core/src/main/java/com/adyen/checkout/components/ui/view/AdyenComponentView.kt

```diff
@@ -58,16 +58,16 @@ class AdyenComponentView @JvmOverloads constructor(
     ) where T : ViewableComponent, T : Component<*, *> {
         component.viewFlow
             .filterNotNull()
-            .onEach {
+            .onEach { componentViewType ->
                 val delegate = component.delegate
```

Comment:
I don't think this works now. The view flow will emit a new view type for the actions but the delegate will always be the CardDelegate. We need a way to return different delegates based on where we are in the flow.

The way we achieve it now in `GenericActionDelegate` is by making the `delegate` change with every new action and overriding the getter:
```kotlin
    private var _delegate: ActionDelegate? = null
    override val delegate: ActionDelegate get() = requireNotNull(_delegate)
```
Not sure if we can do something similar to payment components.

Reply:
The only thing I can come up with is this, but it's not so pretty
```Kotlin
private var _delegate: ComponentDelegate = cardDelegate
override val delegate: ComponentDelegate get() = _delegate
override val viewFlow: Flow<ComponentViewType?> = merge(
        cardDelegate.viewFlow,
        genericActionDelegate.viewFlow,
    )
        .onEach {
            _delegate = when (it) {
                is CardComponentViewType -> cardDelegate
                else -> genericActionDelegate
            }
        }
```

Reply:
What about this?

```kotlin
    private var _delegate: ComponentDelegate = cardDelegate
    override val delegate: ComponentDelegate get() = _delegate

    override fun handleAction(action: Action, activity: Activity) {
        genericActionDelegate.handleAction(action, activity)
        _delegate = genericActionDelegate.delegate
    }
```

But the `handleAction` code is in the `DefaultActionHandlingComponent` so dunno how to achieve that.

Reply:
Also either way, we might have a problem if the view flow emits before the value of `delegate` changes, because he new view will be created but the old delegate will be passed to it :/

Reply:
With what I posted earlier that can't happen, but your solution looks cleaner. Let me test it